src/main/java/com/epam/test_generator/services/TagService.java

```diff
@@ -40,29 +44,29 @@
     @Autowired
     private SuitTransformer suitTransformer;
 
-    @Autowired
-    private TagTransformer tagTransformer;
+	@Autowired
+	private CaseTransformer caseTransformer;
+
+	@Autowired
+	private TagTransformer tagTransformer;
 
     @Autowired
     private CaseVersionDAO caseVersionDAO;
 
-    public Set<TagDTO> getAllTagsFromAllCasesInSuit(Long suitId) {
-        Suit suit = suitDAO.findOne(suitId);
-        checkNotNull(suit);
+	public Set<TagDTO> getAllTagsFromAllCasesInSuit(Long suitId) {
+		Suit suit = suitTransformer.fromDto(suitService.getSuit(suitId));
 
-        Set<TagDTO> allTagsFromAllCases = new HashSet<>();
-        suit.getCases().forEach(caze -> caze.getTags()
-            .forEach(tag -> allTagsFromAllCases.add(tagTransformer.toDto(tag))));
+		Set<TagDTO> allTagsFromAllCases = new HashSet<>();
+		suit.getCases().forEach(caze -> caze.getTags()
```

Comment:
Let's use flatMap instead of forEach in forEach:
[https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#flatMap-java.util.function.Function-](url)

Reply:
Thank you for advice. Done!